### PR TITLE
[G2M] Tooltip - set notitle as string; add PropTypes.number to label prop

### PR DESCRIPTION
**Pull request by geoerika**
**Fixes:**

1. Set `notitle` prop in Tooltip as `string `and use it as such; added number prop type to `label`
2. Fix `baseColors` var in charts to work properly with `colors` as function
3. Add story for the above case
3. Add `id `key to `legend `var so we have an indexing key
4. Added `object `prop type to `axisBottomLabelValues` for when used with dates

### CODE REVIEW

File: src/components/tooltip/index.js [hunk 1]
```diff
@@ -77,7 +80,7 @@ const Tooltip = ({
         </TooltipLabel>
       </TooltipHeader>
     ) }
-    <TooltipBody notitle={ disableTooltipTitle }>
+    <TooltipBody notitle={ disableTooltipTitle.toString() }>
```
Comment (woozyking) [inline src/components/tooltip/index.js:83]: what's the reason for doing this?

Comment (geoerika) [reply src/components/tooltip/index.js:83]: There was a warning in the browser about notitle being boolean. So I switch to work with it as string...maybe something with goober here? This is where I use notitle...

Comment (woozyking) [reply src/components/tooltip/index.js:83]: perhaps goober, should just suppress. it's semantically weirder than the warning itself to convert boolean to string for string comparison with `=== 'true'`

Comment (geoerika) [reply src/components/tooltip/index.js:83]: How do you suppress it? I noticed that if I do this 
` <TooltipBody notitle={ disableTooltipTitle.toString() }>`
the following worked as if notitle was boolean:
`margin-left: ${({ notitle }) => notitle ? '0' : '14px'};`
I guess I can try use it like this...?

**Review by DoParkEQ — APPROVED** [review]

File: src/components/tooltip/index.js (hunk 1 above)
Comment (geoerika) [reply src/components/tooltip/index.js:83]: @woozyking Can you check the last version?

**Review by woozyking — APPROVED** [review]

**Pull request merged**